gallery/__init__.py

```diff
@@ -876,21 +959,22 @@ def get_dir_children(dir_id: int) -> Any:
 @app.route("/api/directory/get/<int:dir_id>")
 @auth.oidc_auth('default')
 @gallery_auth
-def display_files(dir_id: int, internal: bool = False, auth_dict: Optional[Dict[str, Any]] = None):
+def display_files(dir_id: int, internal: bool = False, auth_dict: Optional[Dict[str, Any]] = None, pinned = False):
     gallery_lockdown = util.get_lockdown_status()
     if gallery_lockdown and (not auth_dict['is_eboard'] and not auth_dict['is_rtp']):
         abort(405)
 
-    file_list = [("File", f) for f in File.query.filter(File.parent == dir_id).all()]
-    dir_list = [("Directory", d) for d in Directory.query.filter(Directory.parent == dir_id).all()]
+    file_list = [("File", f) for f in File.query.filter(File.parent == dir_id, File.pinned == pinned).all()]
+    dir_list = [("Directory", d) for d in Directory.query.filter(Directory.parent == dir_id, Directory.pinned == pinned).all()]
```

Comment:
Wait won't this result in not displaying pinned things if pinned isn't true?

Reply:
Anything that is pinned will have pinned true.

Reply:
Can you add a doc comment to indicate that pinned means "only return pinned/unpinned"?